portal-impl/src/com/liferay/portal/action/UpdatePasswordAction.java

```diff
@@ -302,9 +305,31 @@ else if (authType.equals(CompanyConstants.AUTH_TYPE_ID)) {
 			login = String.valueOf(userId);
 		}
 
-		AuthenticatedSessionManagerUtil.login(
-			httpServletRequest, httpServletResponse, login, password1, false,
-			null);
+		try {
+			boolean currentValidate = isValidatePassword(httpServletRequest);
+
+			PwdToolkitUtilThreadLocal.setValidate(currentValidate);
+
+			User userUpdated = UserLocalServiceUtil.updatePassword(
+				userId, password1, password2, passwordReset);
+
+			AuthenticatedSessionManagerUtil.login(
+				httpServletRequest, httpServletResponse, login, password1,
+				false, null);
+
+			UserLocalServiceUtil.sendPasswordNotification(
+				userUpdated, userUpdated.getCompanyId(), password1, null, null,
+				null, null, null);
+		}
+		finally {
+			PwdToolkitUtilThreadLocal.setValidate(previousValidate);
+		}
+
+		if (ticket != null) {
+			TicketLocalServiceUtil.deleteTicket(ticket);
+
+			UserLocalServiceUtil.updatePasswordReset(userId, false);
+		}
```

Comment:
Ticket has already been deleted to this point by Line 284-292 even if the password change was unsuccessful. You might need to set `UserLocalServiceUtil.updatePasswordReset(userId, false)` for successful login but you should set it back if it failed.